process_scheduling_algorithms/non_preemptive_priority_scheduling.c

```diff
@@ -0,0 +1,419 @@
+/**
+ * @file
+ * @brief
+ * [Non-Preemptive Priority
+ * Scheduling](https://en.wikipedia.org/wiki/Scheduling_(computing))
+ * is a scheduling algorithm that selects the tasks to execute based on
+ * priority.
+ *
+ * @details
+ * In this algorithm, processes are executed according to their
+ * priority. The process with the highest priority is to be executed first and
+ * so on. In this algorithm, a variable is maintained known as the time quantum.
+ * The length of the time quantum is decided by the user. The process which is
+ * being executed is interrupted after the expiration of the time quantum and
+ * the next process with the highest priority is executed. This cycle of
+ * interrupting the process after every time quantum and resuming the next
+ * process with the highest priority continues until all the processes have
+ * been executed.
+ * @author [Aryan Raj](https://github.com/aryaraj132)
+ */
+#include <assert.h>   /// for assert
+#include <stdbool.h>  /// for boolean data type
+#include <stdio.h>    /// for IO operations (`printf`)
+#include <stdlib.h>  /// for memory allocation eg: `malloc`, `realloc`, `free`, `exit`
+/**
+ * @brief Structure to represent a process
+ */
+struct node
+{
+    int ID;             ///< ID of the process node
+    int AT;             ///< Arrival Time of the process node
+    int BT;             ///< Burst Time of the process node
+    int priority;       ///< Priority of the process node
+    int CT;             ///< Completion Time of the process node
+    int WT;             ///< Waiting Time of the process node
+    int TAT;            ///< Turn Around Time of the process node
+    struct node *next;  ///< pointer to the node
+};
```

Comment:
This is non-blocking but I think you can use typedef to make it a bit easier to work with

```suggestion
 typedef struct node {
     int ID;             ///< ID of the process node
     int AT;             ///< Arrival Time of the process node
     int BT;             ///< Burst Time of the process node
     int priority;       ///< Priority of the process node
     int CT;             ///< Completion Time of the process node
     int WT;             ///< Waiting Time of the process node
     int TAT;            ///< Turn Around Time of the process node
     struct node *next;  ///< pointer to the node
 } node;
```

then in function definitions, its simply `void insert(node **root, int id, int at, int bt, int prior)` and so on.